frontend/simple/js/state.js

```diff
@@ -188,17 +244,23 @@ const actions = {
       // Subscribe to your fellow group member's identity contracts upon joining
       // TODO right an opposite series of operations for when someone leaves a group
       if (entry instanceof Events.AcceptInvitation && entry.data.username !== state.loggedIn) {
-        let identity = await backend.lookup(entry.data.username)
+        let identity = await namespace.lookup(entry.data.username)
         await backend.subscribe(identity)
-        await dispatch('synchronize', identity)
+        await dispatch('syncContractWithServer', identity)
+      }
+      // TODO Remove this when we have a system for limiting whom we receive messages from
+      if (entry instanceof Events.PostMessage && entry.data.from) {
+        let identity = await namespace.lookup(entry.data.from)
+        if (!state.contracts[identity]) {
+          await backend.subscribe(identity)
+          await dispatch('syncContractWithServer', identity)
+        }
       }
       // TODO: verify each entry is signed by a group member
-      let hash = await db.addLogEntry(contractId, entry)
-      // If this is a duplicate the hash will be null
-      // This may occur if we get duplicate events over the network
-      if (!hash) { return }
-      commit(`${contractId}/${type}`, entry.data)
-
+      await db.addLogEntry(contractId, entry)
+      // If this is a duplicate the hash will be null but we must process further due to users who share the same
+      // machine sharing data storage
```

Comment:
Before we were exiting if the hash came back null and not committing to the state. A null state happens when we have already stored the event. This is wrong for users who are sharing the same storage because the state needs to be updated even if the data is already present. This comment is to explain that.